### PR TITLE
Feature/course docs stats

### PR DESCRIPTION
**Pull request by philratcliffe**
### What

The stats is a large story so breaking it up into smaller PRs. This one covers continuation stats.

### How to review

Review changes. Run on azure with the full hesa data set and ensure creates course entries with expected stats.

### CODE REVIEW

File: EtlPipelineBlobTrigger/dev-tools/process_subj_codes_csv.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+"""
+Create json lookup files for subject code to english and subject code
```
Comment (nshumoogum) [inline EtlPipelineBlobTrigger/dev-tools/process_subj_codes_csv.py]: Is it worth adding a readme for `dev-tools` folder?

File: EtlPipelineBlobTrigger/course_stats.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+import unicodedata
+from collections import OrderedDict
+
+
+class Continuation:
+    """Extracts and transforms the continuation course element"""
+
+    def __init__(self):
+        self.cwd = os.path.dirname(os.path.abspath(__file__))
+        self.subj_code_english = self.get_lookup('subj_code_english')
```
[the remark this answers is not in the document]
Comment (philratcliffe) [reply EtlPipelineBlobTrigger/course_stats.py]: Yes, as discussed this is happening in next story.

File: EtlPipelineBlobTrigger/course_stats.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+            "CONTAGG": 'aggregation_level',
+            "CONTSBJ": 'subject',
+            "UCONT": 'continuing_with_provider',
+            "UDORMANT": 'dormant',
+            "UGAINED": 'gained',
+            "ULEFT": 'left',
+            "ULOWER": 'lower',
+        }[xml_key]
+
+    def get_lookup(self, lookup_name):
```
[the remark this answers is not in the document]
Comment (philratcliffe) [reply EtlPipelineBlobTrigger/course_stats.py]: I wanted to create an abstraction so the caller didn't have to be aware of how the lookups were stored. That is the location, or even the type of storage - e.g., file, azure storage, DB etc.

File: EtlPipelineBlobTrigger/course_stats.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+                                cont_elem)
+                else:
+                    continuation[json_key] = cont_elem[xml_key]
+                ordered_continuation = OrderedDict(sorted(
+                    continuation.items()))
+            continuation_list.append(ordered_continuation)
+        return continuation_list
+
+
+class CourseStats:
```
[the remark this answers is not in the document]
Comment (philratcliffe) [reply EtlPipelineBlobTrigger/course_stats.py]: OK will do in next story as part of the refactor.

File: EtlPipelineBlobTrigger/dev-tools/process_subj_codes_csv.py (hunk 1 above)
Comment (philratcliffe) [reply EtlPipelineBlobTrigger/dev-tools/process_subj_codes_csv.py]: Yes, makes sense. Done.

**Pull request merged**